nntrainer/layers/mol_attention_layer.cpp

```diff
@@ -24,18 +25,103 @@ MoLAttentionLayer::~MoLAttentionLayer() {}
 
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 
-enum AttentionParams { query = 0, value = 1, key = 2, score, weights };
+enum AttentionParams { query = 0, value = 1, state = 2, mlp_w, mlp_proj_w };
 
 void MoLAttentionLayer::finalize(InitLayerContext &context) {
-  finalizeCommon(context);
-  /** NYI */
+  if (context.getNumInputs() != 3)
+    throw std::runtime_error("MoL Attention layer needs 3 inputs.");
+
+  auto const &all_dims = context.getInputDimensions();
+  auto const &query_dim = all_dims[AttentionParams::query];
+
+  wt_idx[AttentionParams::query] = AttentionParams::query;
+  wt_idx[AttentionParams::value] = AttentionParams::value;
+  wt_idx[AttentionParams::state] = AttentionParams::state;
+
+  softmax.setActiFunc(ActivationType::ACT_SOFTMAX);
+  tanh.setActiFunc(ActivationType::ACT_TANH);
+  sigmoid.setActiFunc(ActivationType::ACT_SIGMOID);
+
+  auto unit = std::get<props::Unit>(mol_props).get();
+  auto mol_k = std::get<props::MoL_K>(mol_props).get();
+
+  TensorDim mlp_w_dim = {query_dim.width(), unit};
+  wt_idx[AttentionParams::mlp_w] =
+    context.requestWeight(mlp_w_dim, Tensor::Initializer::XAVIER_UNIFORM,
```

Comment:
wonder if the initializers are fixed?

Reply:
Its updated to use LayerImpl in the upcoming PR.